### PR TITLE
Enable query sharding in Helm

### PR DESCRIPTION
**Pull request by dimitarvdimitrov**
Signed-off-by: Dimitar Dimitrov <dimitar.dimitrov@grafana.com>

#### What this PR does

Modifies the mimir config in the Helm chart to enable query sharding for all tenants.

#### Which issue(s) this PR fixes or relates to

Fixes https://github.com/grafana/mimir/issues/2594

#### Checklist

- [ ] Tests updated
- [ ] Documentation added
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: operations/helm/charts/mimir-distributed/CHANGELOG.md [hunk 1]
```diff
@@ -17,6 +17,7 @@ Entries should include a reference to the Pull Request that introduced the chang
   - Renamed helm config values `minio.accessKey` to `minio.rootUser`.
   - Renamed helm config values `minio.secretKey` to `minio.rootPassword`.
   - Minio container images are now loaded from quay.io instead of Docker Hub. Set `minio.image.repository` value to override the default behavior.
+* [CHANGE] Enable [query sharding](https://grafana.com/docs/mimir/latest/operators-guide/architecture/query-sharding/) by default. If you override the value of `mimir.config`, then take a look at `mimir.config` in the `values.yaml` from this version of the chart and incorporate the differences. If you override `mimir.config`, then consider switching to `mimir.structuredConfig`. #2655
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/CHANGELOG.md:20]: Let's mention that setting `mimir.structuredConfig.parallelize_shardable_queries` to `false` can be used to turn off query sharding.

Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/CHANGELOG.md]: added and rebased onto the lastest main. PTAL

File: operations/helm/charts/mimir-distributed/values.yaml [hunk 2]
```diff
@@ -179,6 +179,7 @@ mimir:
       {{- if .Values.query_scheduler.enabled }}
       scheduler_address: {{ template "mimir.fullname" . }}-query-scheduler-headless.{{ .Release.Namespace }}.svc:{{ include "mimir.serverGrpcListenPort" . }}
       {{- end }}
+      parallelize_shardable_queries: true
```
Comment (krajorama) [inline operations/helm/charts/mimir-distributed/values.yaml:182]: should go before the scheduler_address to have same alphabetic order as in the generated config

Comment (krajorama) [inline operations/helm/charts/mimir-distributed/values.yaml:182]:
```suggestion
      parallelize_shardable_queries: true
      {{- if .Values.query_scheduler.enabled }}
      scheduler_address: {{ template "mimir.fullname" . }}-query-scheduler-headless.{{ .Release.Namespace }}.svc:{{ include "mimir.serverGrpcListenPort" . }}
      {{- end }}
```

Comment (dimitarvdimitrov) [reply operations/helm/charts/mimir-distributed/values.yaml]: done in https://github.com/grafana/mimir/pull/2655/commits/e5b2b1a7dabf47d8190d052726e5bdef9be763ee

**Review by krajorama — APPROVED** [review]
LGTM

**Pull request merged**